micro-rdk/src/common/credentials_storage.rs

```diff
@@ -174,46 +195,82 @@ impl RAMStorage {
     }
-    fn store_ota_metadata(&self, ota_metadata: OtaMetadata) -> Result<(), Self::Error> {
+    fn store_ota_metadata(&self, ota_metadata: &OtaMetadata) -> Result<(), Self::Error> {
         let mut inner_ref = self.0.lock().unwrap();
-        let _ = inner_ref.ota_metadata.insert(ota_metadata);
+        let _ = inner_ref.ota_metadata.insert(ota_metadata.clone());
         Ok(())
     }
     fn get_ota_metadata(&self) -> Result<OtaMetadata, Self::Error> {
         let inner_ref = self.0.lock().unwrap();
-        Ok(inner_ref.ota_metadata.clone().unwrap_or_default())
+        inner_ref
+            .ota_metadata
+            .clone()
+            .ok_or(RAMStorageError::NotFound)
     }
     fn reset_ota_metadata(&self) -> Result<(), Self::Error> {
         let _ = self.0.lock().unwrap().ota_metadata.take();
         Ok(())
     }
 }
+#[cfg(feature = "ota")]
+impl<Iterable, Storage: OtaMetadataStorage> OtaMetadataStorage for Iterable
+where
+    for<'a> &'a Iterable: IntoIterator<Item = &'a Storage>,
+    Storage::Error: From<EmptyStorageCollectionError>,
+{
+    type Error = Storage::Error;
+    fn has_ota_metadata(&self) -> bool {
+        self.into_iter().any(OtaMetadataStorage::has_ota_metadata)
+    }
+    fn get_ota_metadata(&self) -> Result<OtaMetadata, Self::Error> {
+        self.into_iter().fold(
+            Err::<_, Self::Error>(EmptyStorageCollectionError.into()),
+            |val, s| val.or_else(|_| s.get_ota_metadata()),
+        )
+    }
+    fn reset_ota_metadata(&self) -> Result<(), Self::Error> {
+        self.into_iter().fold(
+            Err::<_, Self::Error>(EmptyStorageCollectionError.into()),
+            |val, s| val.or(s.reset_ota_metadata()),
+        )
+    }
+    fn store_ota_metadata(&self, ota_metadata: &OtaMetadata) -> Result<(), Self::Error> {
+        self.into_iter().fold(
+            Err::<_, Self::Error>(EmptyStorageCollectionError.into()),
+            |val, s| val.or_else(|_| s.store_ota_metadata(ota_metadata)),
+        )
+    }
+}
 
 impl RobotConfigurationStorage for RAMStorage {
-    type Error = <RobotCredentials as TryFrom<CloudConfig>>::Error;
+    type Error = RAMStorageError;
     fn has_robot_credentials(&self) -> bool {
         let inner_ref = self.0.lock().unwrap();
         inner_ref.robot_creds.is_some()
     }
-    fn store_robot_credentials(&self, cfg: CloudConfig) -> Result<(), Self::Error> {
-        let creds: RobotCredentials = cfg.try_into()?;
+    fn store_robot_credentials(&self, cfg: &CloudConfig) -> Result<(), Self::Error> {
+        let creds: RobotCredentials = cfg.clone().try_into()?;
         let mut inner_ref = self.0.lock().unwrap();
         let _ = inner_ref.robot_creds.insert(creds);
         Ok(())
     }
     fn get_robot_credentials(&self) -> Result<RobotCredentials, Self::Error> {
         let inner_ref = self.0.lock().unwrap();
-        let cfg = inner_ref.robot_creds.clone().unwrap_or_default().clone();
-        Ok(cfg)
+        log::info!("get robot creds");
```

Comment:
Stray log, or `debug!`?

As an aside, I do miss some of the logging we used to have pre #316 that gave a little more context about progression through the startup sequence, whether we were using cached credentials or configs, etc. I'm usually for "silence is golden" but in this case that's very useful information to understand how things are coming up, especially in offline mode. Would you be OK with a ticket to re-introduce some of that logging? Or did you not propagate into the refactored world because you found it too chatty?

Reply:
stray log i will remove it.
Yes let's make a ticket for some logging housekeeping

Reply:
Done, in https://viam.atlassian.net/browse/RSDK-9715. Feel free to add more things you'd like to see added to a logging pass.